### ISSUE TITLE
Illegal code on Colab

### ISSUE THREAD

**Issue by ll2pakll** [TheLastBen/fast-stable-diffusion#2013]
When I tried to run this Notebook, I started getting a message saying that this code is not allowed on Colab and if I continue to run it, my rights to use Colab will be restricted. Has anyone else received a message like this? This is the first time I've received it, although I've used this Notebook before with no problems

**Comment by LukeOvermind** [comment]
Everything was fin an hour ago, took a break to get the kid ready for bed and now I am getting "You may be executing code that is disallowed, and this may restrict your ability to use Colab in the future. Please note the prohibited actions specified in our [FAQ](https://research.google.com/colaboratory/faq.html#limitations-and-restrictions)."

**Comment by Larry2600** [comment]
Got the same popup.

**Comment by InternalMegaT** [comment]
Same

**Comment by lyery** [comment]
running remote UIs now is against colab TOS 
![imagen](https://user-images.githubusercontent.com/98701013/233451860-6927d86e-3a9a-4223-aef4-2aede9168581.png)

**Comment by MartialWrapperr** [comment]
It's over. Google is cracking down on poor people.

**Comment by InternalMegaT** [comment]
> running remote UIs now is against colab TOS edit: this includes SD, of course ![imagen](https://user-images.githubusercontent.com/98701013/233451860-6927d86e-3a9a-4223-aef4-2aede9168581.png)

This also applies to paid units?

**Comment by Larry2600** [comment]
Leave it to Google to ruin everything. I imagine a lot of people only pay for Collab just for A1111. Absolutely stupid.

**Comment by InternalMegaT** [comment]
> running remote UIs now is against colab TOS edit: this includes SD, of course ![imagen](https://user-images.githubusercontent.com/98701013/233451860-6927d86e-3a9a-4223-aef4-2aede9168581.png)

Does this also apply to Dreambooth colab?

**Comment by BannedMan** [comment]
> Leave it to Google to ruin everything. I imagine a lot of people only pay for Collab just for A1111. Absolutely stupid.

I mean free gpu's and ram.... be lucky we got it in the 1st place, im a bit peeved tho

**Comment by InternalMegaT** [comment]
> > Leave it to Google to ruin everything. I imagine a lot of people only pay for Collab just for A1111. Absolutely stupid.
> 
> I mean free gpu's and ram.... be lucky we got it in the 1st place, im a bit peeved tho

So this does only apply to free units?

**Comment by BannedMan** [comment]
> > > Leave it to Google to ruin everything. I imagine a lot of people only pay for Collab just for A1111. Absolutely stupid.
> > 
> > 
> > I mean free gpu's and ram.... be lucky we got it in the 1st place, im a bit peeved tho
> 
> So this does only apply to free units?

Idk, maybe they banned it outright

**Comment by lyery** [comment]
> > running remote UIs now is against colab TOS edit: this includes SD, of course ![imagen](https://user-images.githubusercontent.com/98701013/233451860-6927d86e-3a9a-4223-aef4-2aede9168581.png)
> 
> This also applies to paid units?

Yes, this is also applies to paid units/colab pro

**Comment by InternalMegaT** [comment]
> > > running remote UIs now is against colab TOS edit: this includes SD, of course ![imagen](https://user-images.githubusercontent.com/98701013/233451860-6927d86e-3a9a-4223-aef4-2aede9168581.png)
> > 
> > 
> > This also applies to paid units?
> 
> Yes, this is also applies to paid units/colab pro

WHAT, ok I understand on free units bit paid units aswell!!

**Comment by InternalMegaT** [comment]
@TheLastBen Maybe there is a way to fix this? Hopefully. Does this also affect your Dreambooth colab?

**Comment by Larry2600** [comment]
> @TheLastBen Maybe there is a way to fix this? Hopefully. Does this also affect your Dreambooth colab?

I doubt Ben can do anything, it's global.  Might be the end.

**Comment by InternalMegaT** [comment]
> > @TheLastBen Maybe there is a way to fix this? Hopefully. Does this also affect your Dreambooth colab?
> 
> I doubt Ben can do anything, it's global. Might be the end.

It says it blocked remote ui's, remote desktops, etc. There may be a way to not use that.

**Comment by lgooders** [comment]
I have colab pro and I am getting the same message.

**Comment by BannedMan** [comment]
> I have colab pro and I am getting the same message.

its a TOS, not a teir gate

**Comment by Larry2600** [comment]
> I have colab pro and I am getting the same message.

It applies to everyone, regardless if paid or free version.

**Comment by Trokinos** [comment]
So i assume it means that google does not want anymore to allow people to use the excellent  A1111 with a google colab ?

**Comment by OswaldoInfante97** [comment]
> So i assume it means that google does not want anymore to allow people to use the excellent A1111 with a google colab ?

Yeah, it applies to any UI whatsoever. so... no point in paying that shit now

**Comment by ralph2121** [comment]
NovelAI(4chan_Ver_)/Naifu Diffusion if it is working.


Shouldn't that one also be failing?

**Comment by materosx** [comment]
I've read on reddit that the warning only appears if the words "stable diffusion" are somewhere inside the code, so maybe just removing/replacing all of them would fix it? 
![kff4138305va1](https://user-images.githubusercontent.com/104952968/233497030-7c1a0bf8-546f-415d-8d19-e5244796117f.png)

**Comment by InternalMegaT** [comment]
> I've read on reddit that the warning only appears if the words "stable diffusion" are somewhere inside the code, so maybe just removing/replacing all of them would fix it? 
> ![kff4138305va1](https://user-images.githubusercontent.com/104952968/233497030-7c1a0bf8-546f-415d-8d19-e5244796117f.png)
> 
> 

There's a fix hope this is true and can be changed/fixed

**Comment by phoenix9990** [comment]
> > I've read on reddit that the warning only appears if the words "stable diffusion" are somewhere inside the code, so maybe just removing/replacing all of them would fix it?
> > ![kff4138305va1](https://user-images.githubusercontent.com/104952968/233497030-7c1a0bf8-546f-415d-8d19-e5244796117f.png)
> 
> There's a fix hope this is true and can be changed/fixed

I mean it would still be short live if there is any such fix Google just like that

**Comment by letrasderosa** [comment]
> 

The same with me.

**Comment by Gushousekai195** [comment]
What's the alternative then?!! Goddammit to Hell!!

I still can't afford a GTX 3090 with 24gb to run it locally!! And I don't see it happening anytime soon!! I've not been able to save my money much lately....

**Comment by Gushousekai195** [comment]
https://www.reddit.com/r/StableDiffusion/comments/12t8tc7/is_colab_going_to_start_banning_people_who_use_it/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button

**Comment by TheLastBen** [comment]
There are alternative fast-stable-diffusion notebooks for Runpod and Paperspace

**Comment by Gushousekai195** [comment]
> There are alternative fast-stable-diffusion notebooks for Runpod and Paperspace

Can I load my Google Drive on them?

**Comment by TheLastBen** [comment]
@Gushousekai195 you can share the folder on your gdrive, then use "gdown --folder" to download it

**Comment by qa13683954882** [comment]
Silly bitch Google, he thought he would make more people pay to choose the pro model, but not so, it will only make most people abandon Colab and choose other GPU platforms, the funniest thing is that the subtle rules still apply to pro users, instead you pay and do not feel safe and may be interfered with at any time.

**Comment by RebelCat90** [comment]
I have Colab PRO and I also have problems with that message.
I've done some work still, I don't know how risky it could be that they won't give me access later or ban my account.

Excuse my English, I'm Latin American

**Comment by Sufyan-007** [comment]
I think I got a work around, for now. Apparently, all they're doing is searching for "stable-diffsion-webui" (in text format) in your code which is executed as a command line instruction (cloning the repo) . I stored "stable-diffsion-webui" in a variable (which doesn't trigger a warning on its own), and replaced it wherever there was a "stable-diffsion-webui" written in command line code. This is still technically against the TOS, but the warning goes away

**Comment by RebelCat90** [comment]
> 

I also tried to edit the text where "stable diffusion" appeared and when I get to the ControlNet section it sends me an error that says it can't find the models

**Comment by RebelCat90** [comment]
> I think I got a work around, for now. Apparently, all they're doing is searching for "stable-diffusion-webui" (in text format) in your code which is executed as a command line instruction (cloning the repo) . I stored "stable-diffusion-webui" in a variable (which doesn't trigger a warning on its own), and replaced it wherever there was a "stable-diffusion-webui" written in command line code. This is still technically against the TOS, but the warning goes away
> 
> Edit: I made a typo in the code, turns out, You cannot store "stable-diffusion-webui" as a variable either. You simple cannot have the text stable-diffsion-webui anywhere in your code. Nevertheless, simply splitting that and appending it back, s="stable-diffusion"+"-webui" isn't illegal
> 
> after this, Edit -> find and replace -> search stable-diffusion-webui and replace with $s

I also tried to edit the text where "stable diffusion" appeared and when I get to the ControlNet section it sends me an error that says it can't find the models

**Comment by Sufyan-007** [comment]
> > I think I got a work around, for now. Apparently, all they're doing is searching for "stable-diffusion-webui" (in text format) in your code which is executed as a command line instruction (cloning the repo) . I stored "stable-diffusion-webui" in a variable (which doesn't trigger a warning on its own), and replaced it wherever there was a "stable-diffusion-webui" written in command line code. This is still technically against the TOS, but the warning goes away
> > Edit: I made a typo in the code, turns out, You cannot store "stable-diffusion-webui" as a variable either. You simple cannot have the text stable-diffsion-webui anywhere in your code. Nevertheless, simply splitting that and appending it back, s="stable-diffusion"+"-webui" isn't illegal
> > after this, Edit -> find and replace -> search stable-diffusion-webui and replace with $s
> 
> I also tried to edit the text where "stable diffusion" appeared and when I get to the ControlNet section it sends me an error that says it can't find the models

I just tried using control-net and it worked for me. Check if the CN models are stored in the right directory

**Comment by TheLastBen** [comment]
https://github.com/TheLastBen/fast-stable-diffusion/commit/a755c66f8d6a90d8071bf23300540108ae036d69

**Comment by qa13683954882** [comment]
Yes, I tested the code about colab with stable-diffusion-webui in it and all other related plugins are now not disabled.
Interestingly enough, if you change stable-diffusion-webui to →stable diffusion webu, you won't get an error, so maybe you can use a file to cover up the error.

**Comment by TheLastBen** [comment]
use the latest notebook

**Comment by RebelCat90** [comment]
I would appreciate if you share any screenshots about the possible solutions

**Comment by TheLastBen** [comment]
@RebelCat90 Use the latest notebook and you won't get any message or error

**Comment by Gushousekai195** [comment]
I also heard that ngrok is banned on Colab

**Comment by TheLastBen** [comment]
ngrok is just like gradio server, you can use it

**Comment by qa13683954882** [comment]
> @RebelCat90使用最新的笔记本，你不会收到任何消息或错误

Thanks brother hope Google does not commit any disease.

**Comment by lyery** [comment]
> ngrok is just like gradio server, you can use it

you should put a warning in the collab telling people that using this now is against google's tos and anyone who uses this can possibly be banned

**Comment by Tirtho911** [comment]
![image](https://user-images.githubusercontent.com/94674322/233558031-d5aaa87a-e8a4-44b2-af65-21f01872d78d.png)
what is this

**Comment by Archon332** [comment]
> use the latest notebook

I deleted old SD and reuploaded new SD version using link from from this Github page. There was no warning while uploading but after during uploading of extensions warning with illegal bullshit apeared once again.

**Comment by Archon332** [comment]
I understand something. Warning appears when you tray to add path to your model. 
So, shen I started UI with usual SD model there was no warning and it was when I tried to add "/content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/grapefruitHentaiModel_grapefruitv4.safetensors" to the path line.

**Comment by Stoi84** [comment]
Not getting any message with an older colab (the one just before the fix). I have Colab Pro, from France.

**Comment by InternalMegaT** [comment]
> Not getting any message with an older colab (the one just before the fix). I have Colab Pro, from France.

Are you saying they revised the TOS?

**Comment by Stoi84** [comment]
I'm just sharing my own experience in case it's of any help. I might not getting the message for other reasons.

**Comment by TheLastBen** [comment]
Pro users can use it without restrictions

**Comment by Stoi84** [comment]
Some mentioned earlier here that even with Pro, they're getting this message. Thanks for the clarification about this.

**Comment by InternalMegaT** [comment]
> Pro users can use it without restrictions 

Im confused there were so many pro users having this issue. Are you 100% sure?

**Comment by TheLastBen** [comment]
> Im confused there were so many pro users having this issue. Are you 100% sure?

Yes

**Comment by Larry2600** [comment]
> > Pro users can use it without restrictions
> 
> Im confused there were so many pro users having this issue. Are you 100% sure?

Looks like pro users will still be able to use it. Source: https://twitter.com/thechrisperry/status/1649196140909428736

Since you are a paid user you should be good. Rest of us plebians from second/third world countries that can't afford the subscription or a good PC are screwed.

**Comment by InternalMegaT** [comment]
> > > Pro users can use it without restrictions
> > 
> > Im confused there were so many pro users having this issue. Are you 100% sure?
> 
> Looks like pro users will still be able to use it. Source: https://twitter.com/thechrisperry/status/1649196140909428736
> 
> Since you are a paid user you should be good. Rest of us plebians from second/third world countries that can't afford the subscription or a good PC are screwed. 
> 
> 

Do if you're in a first world country and using free units it will be fine?

**Comment by Larry2600** [comment]
> > > > Pro users can use it without restrictions
> > > 
> > > 
> > > Im confused there were so many pro users having this issue. Are you 100% sure?
> > 
> > 
> > Looks like pro users will still be able to use it. Source: https://twitter.com/thechrisperry/status/1649196140909428736
> > Since you are a paid user you should be good. Rest of us plebians from second/third world countries that can't afford the subscription or a good PC are screwed.
> 
> So if you're in a first world country and using free units it will be fine?

No, everyone will have to use paid units. At least that's what the colab lead is saying in that tweet I linked. It's not set in stone yet but they're most likely going to pull the plug on free tier SD Webui. 

If you're a paid user you can use the WebUI to your heart's content without the worry of being banned.

**Comment by infernus4444** [comment]
> When I tried to run this Notebook, I started getting a message saying that this code is not allowed on Colab and if I continue to run it, my rights to use Colab will be restricted. Has anyone else received a message like this? This is the first time I've received it, although I've used this Notebook before with no problems

In nocrypt's colab, unchecking output to drive solves the problem

**Comment by profcat2** [comment]
Not mounting google drive kinda works, the webui changes a bit and gradio launches and it starts to produce an image but then the final render is just a teal square.

**Comment by aefly** [comment]
For those who still have the "Illegale code", move your models to another folder than the one in stable-diffusion-webui:

Example: /content/gdrive/MyDrive/sd/models/

**Comment by plucyvrz** [comment]
> For those who still have the "Illegale code", move your models to another folder than the one in stable-diffusion-webui:
> 
> Example: /content/gdrive/MyDrive/sd/models/

I've tried this, and a new folder is like /content/gdrive/MyDrive/WebUI/models/ but got the same message. Is there any other trick for bypassing that message?

**Comment by aefly** [comment]
@plucyvrz For now it's the only way I found, try without "WebUI" folder, like: /content/gdrive/MyDrive/sd/models/yourmodel.safetensors

**Comment by aleqniko** [comment]
I can't pay Goole Colab My country is not in the list

**Comment by TheLastBen** [comment]
@aleqniko use my notebooks in RunPod or Paperspace 
https://www.runpod.io/console/gpu-secure-cloud?template=runpod-stable-unified
https://console.paperspace.com/github/TheLastBen/PPS?machine=Free-GPU

**Comment by aleqniko** [comment]
> @aleqnikoиспользовать мои блокноты в RunPod или Paperspace [https://www.runpod.io/console/gpu-secure-cloud?template=runpod-stable-unified ](https://www.runpod.io/console/gpu-secure-cloud?template=runpod-stable-unified) [https://console.paperspace.com/github/TheLastBen/PPS?machine= Бесплатный графический процессор](https://console.paperspace.com/github/TheLastBen/PPS?machine=Free-GPU)

Thanks. I will definitely look

**Comment by TheLastBen** [comment]
@aleqniko correction for runpod : https://www.runpod.io/console/gpu-browse?template=runpod-stable-unified

**Comment by aleqniko** [comment]
strange but I managed to run the sd without problems
![1212121](https://user-images.githubusercontent.com/42311041/233778075-f1db1cf1-7942-41b5-88dd-a6cb8cec42f9.png)

**Comment by iqddd** [comment]
RunPod is forcing me pay through this shady crap:
![image](https://user-images.githubusercontent.com/37406615/233780257-17138b69-1a6f-4461-ba03-b5893dca8d48.png)
Their app forces identity verification! Cryptocurrency and identity verification, Karl! How can the two concepts coexist?

**Comment by jjhaggar** [comment]
**Current state of events:**
- It seems that Google [doesn't want people to use remote UIs from colab](https://research.google.com/colaboratory/faq.html#limitations-and-restrictions) (like automatic1111 webUI)
- Since it seems Google detects specifically "stable-diffusion-webui", there are some "[fixes](https://github.com/TheLastBen/fast-stable-diffusion/commit/a755c66f8d6a90d8071bf23300540108ae036d69)" that let you run it, (but _that's still against Google TOS_, so use it at your own risk!)
- The only legal option for using colab to generate images with Stable Diffusion would be by _controlling it from inside the collab_. Which could be done, but it would require too much work if you want to have the capabilities of automatic1111 (you would need to write a UI from scratch in python). And even then it just couldn't be as easy to use or comfortable as auto1111 webUI.

**Alternatives to Google colab:**
- Paid: [Runpod](https://blog.runpod.io/stable-diffusion-ui-on-runpod/) & [Vast.ai](https://vast.ai/docs/guides/stable-diffusion)
- Free: [The AI Horde](https://stablehorde.net/), there is a [Stable Horde client for automatic1111](https://github.com/natanjunges/stable-diffusion-webui-stable-horde)

**Comment by TheLastBen** [comment]
@iqddd is that the only payment option ?

**Comment by iqddd** [comment]
@TheLastBen for me it is.

**Comment by XieBaoshi** [comment]
> **Alternatives to Google colab:**
> 
>     * Paid: [Runpod](https://blog.runpod.io/stable-diffusion-ui-on-runpod/) & [Vast.ai](https://vast.ai/docs/guides/stable-diffusion)
> 
>     * Free: [The AI Horde](https://stablehorde.net/), there is a [Stable Horde client for automatic1111](https://github.com/natanjunges/stable-diffusion-webui-stable-horde)

Kaggle also

**Comment by TheLastBen** [comment]
@iqddd Checkout paperspace https://console.paperspace.com/github/TheLastBen/PPS?machine=Free-GPU

**Comment by Mudereded401** [comment]
> @iqddd Checkout paperspace https://console.paperspace.com/github/TheLastBen/PPS?machine=Free-GPU

Thanks, but how can i add commands into that webpage? I want for example, download VAE or LoRA models into it but idk how to do it. (I did it in colab but idk how that page works)

**Comment by TheLastBen** [comment]
click on the jupyter icon on the left panel to use the jupyter interface, it's easier and similar to colab

**Comment by JackG0T** [comment]
> @plucyvrz For now it's the only way I found, try without "WebUI" folder, like: /content/gdrive/MyDrive/sd/models/yourmodel.safetensors

You saved mi life with this <3

**Comment by JackG0T** [comment]
> For those who still have the "Illegale code" when adding path to model, move your models to another folder than the one in stable-diffusion-webui:
> 
> Example: /content/gdrive/MyDrive/sd/models/

@aefly you saved my life

**Comment by aefly** [comment]
@JackG0T no problem ;)

**Comment by profcat2** [comment]
I'm not getting any illegal code warnings from running the collab anymore, however it doesn't work without paying collab which was the main issue, too many people on the platform. News articles also seem to indicate this is the case. For EU/Asia I would recommend the paperspace or runpod options as other posters suggested

**Comment by Shun-Ge** [comment]
Is this because google has their own AI shit models and are cracking down now?

**Comment by firstvanq** [comment]
> @iqddd Checkout paperspace https://console.paperspace.com/github/TheLastBen/PPS?machine=Free-GPU

hey ben, can i not mount my google drive to paperspace?

**Comment by TheLastBen** [comment]
@firstvanq you can share any gdrive folder, then download it with `!gdown --folder (link)`
